### DISCUSSION TITLE
Use huggingface-embeddings to load local embedding model

### DISCUSSION THREAD

**Discussion by flotos** [mudler/LocalAI#959] [Q&A]
**Is your feature request related to a problem? Please describe.**

I am using the following configuration and a downloaded model as I want to deploy on an instance with no internet access

```
backend: huggingface-embeddings
embeddings: true
name: all-minilm
parameters:
  model: all-MiniLM-L6-v2.bin
```

However I get the following error at inference :
```
{
  "error": {
    "code": 500,
    "message": "could not load model (no success): Unexpected err=RepositoryNotFoundError('401 Client Error. (Request ID: Root=1-64e750c4-1941428a25f0e1956ea10fa1;33620328-90bf-49a5-9e52-01fa29fa39d6)\\n\\nRepository Not Found for url: https://huggingface.co/api/models/sentence-transformers/all-MiniLM-L6-v2.bin.\\nPlease make sure you specified the correct `repo_id` and `repo_type`.\\nIf you are trying to access a private or gated repo, make sure you are authenticated.\\nInvalid username or password.'), type(err)=<class 'huggingface_hub.utils._errors.RepositoryNotFoundError'>",
    "type": ""
  }
}
```

I indeed specified a bin file, and my other models work well so it should in theory look into the correct folder.

**Describe the solution you'd like**

If there is a file extension load a local model, or add a parameter for that.

[a comment by an automated account is not shown]

**Comment by mudler** [comment]
well - actually @localai-bot is correct here. the `huggingface-embeddings` backend wants a huggingface repository in the model name. See the documentation in https://localai.io/features/embeddings/index.html#huggingface-embeddings . moving to a discussion
